tests/unit/fork_test.cpp

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/process.h>
+   int vfs_dup(fs_handle h, fs_handle *dup_h);
```

Comment:
Not 100% sure, but I guess that `vfs_dup()` and `vfs_close()` are not required to be defined there. As mocked methods, they should be defined by `mocking.h`. Could you please check? If they are not required, just remove those definitions from here. Otherwise, let me know.